src/cmd/services/m3dbnode/main/common_test.go

```diff
@@ -139,3 +141,35 @@ func newNamespaceWithIndexProtoValue(id string, indexEnabled bool) (proto.Messag
 	}
 	return namespace.ToProto(nsMap), nil
 }
+
+// waitUntilAllShardsAreAvailable continually polls the session checking to see if the topology.Map
+// that the session is currently storing contains a non-zero number of host shard sets, and if so,
+// makes sure that all their shard states are Available.
+func waitUntilAllShardsAreAvailable(t *testing.T, session client.AdminSession) {
+outer:
```

Comment:
should there be a timeout here too ?

Reply:
Eh, the tests will already timeout by default. Also I'm just moving this function, not adding it

Reply:
yeah fair enough.